lambdas/utils/lloyd_george_validator.py

```diff
@@ -87,3 +96,38 @@ def check_for_file_names_agrees_with_each_other(file_name_list: list[str]):
     ]
     if len(set(expected_common_part)) != 1:
         raise LGInvalidFilesException("File names does not match with each other")
+
+def validate_with_pds_service(file_name_list: list[str], nhs_number: str):
+    try:
+    # Check file names match with the nhs number in metadata.csv
+        file_name_info = extract_info_from_filename(file_name_list[0])
+        if file_name_info["nhs_number"] != nhs_number:
+            raise LGInvalidFilesException(
+                "NHS number in file names does not match the given NHS number"
+            )
+        patient_name = file_name_info["patient_name"]
+        date_of_birth = file_name_info["date_of_birth"]
+
+        pds_service = PdsApiService(SSMService())
+        pds_response = pds_service.pds_request(nsh_number=nhs_number, retry_on_expired=True)
+        pds_response.raise_for_status()
+        patient = Patient.model_validate(pds_response.json())
+        patient_details = patient.get_minimum_patient_details(nhs_number)
+
+        date_of_birth = datetime.datetime.strptime(date_of_birth, '%d-%m-%Y').date()
+        if patient_details.birth_date != date_of_birth:
+            raise LGInvalidFilesException("Patient DoB does not match our records")
+        patient_full_name =' '.join([name for name in patient_details.given_Name]) + ' ' + patient_details.family_name
+        if patient_full_name != patient_name:
+            raise LGInvalidFilesException("Patient name does not match our records")
+        current_user_ods = get_user_ods_code()
+        if patient_details.general_practice_ods != current_user_ods:
+            raise LGInvalidFilesException("User is not allowed to access patient")
```

Comment:
Later maybe it's better to make the `current_user_ods` to an argument that got passed into this method, so that it is easier to extend to other GP. Anyway this is not urgent and can be skip for now.

Reply:
Agree. I think we can leave that for future refactoring.